### Testing accuracy of RF classifier by providing training data of 20Hz, 30Hz and 40Hz defects and testing out it's accuracy on different speed readings of 10Hz, 15Hz, 20Hz, 25HZ, 30Hz and 35Hz. 

### Training Dataset consists of 10000 rows for each fault and testing data consists of 20000 rows for each fault. Mix data consists of 200 rows data for each fault in all rotational speeds.

In [2]:
import pandas as pd

data_20=pd.read_csv(r'D:\Acads\BTP\Train\Train20Hz.csv')
data_30=pd.read_csv(r'D:\Acads\BTP\Train\Train30Hz.csv')
data_40=pd.read_csv(r'D:\Acads\BTP\Train\Train40Hz.csv')

In [3]:
data_20.head

<bound method NDFrame.head of         X_Value  Acceleration  Acceleration_0  Acceleration_1   Voltage  \
0       0.00000      1.708963       22.936699        0.418647  0.062727   
1       0.00005      0.397558       20.744087       -2.443510  0.058105   
2       0.00010      0.245741       19.886220       -4.793004  0.065566   
3       0.00015     -0.492077       16.847528       -7.492809  0.060839   
4       0.00020      1.393773       15.511186       -5.217568  0.068609   
...         ...           ...             ...             ...       ...   
100239  1.01385     -2.401899       -2.835274       -2.798354  0.168155   
100240  1.01390      1.870737       -1.357366        0.724601  0.170059   
100241  1.01395      1.843078       -2.764307        1.164236  0.166506   
100242  1.01400      5.318520        0.023913        4.574955  0.167842   
100243  1.01405      5.640224       -0.904798        2.451036  0.162821   

        Voltage_0  Voltage_1  Voltage_2  Voltage_3  Identity  
0     

In [4]:
data_30.head

<bound method NDFrame.head of         X_Value  Acceleration  Acceleration_0  Acceleration_1   Voltage  \
0      0.000000    -11.367953      -11.637524        6.021533 -0.211798   
1      0.000050    -12.166375       -9.140292        3.120780 -0.217642   
2      0.000100     -8.550057       -4.918858        0.881439 -0.211531   
3      0.000150     -7.510203       -1.977898       -0.244971 -0.217031   
4      0.000200     -9.059351        0.536890        3.866246 -0.210774   
...         ...           ...             ...             ...       ...   
99948  0.999350      6.308956       -7.804564        2.868125 -0.231882   
99949  0.999400      3.970360       -6.623541        1.938864 -0.229132   
99950  0.999450      1.442453       -4.509385        2.251926 -0.233108   
99951  0.999500     -0.867132       -3.078502        0.568868 -0.229825   
99952  0.999528     -0.404975       -0.450460       -0.560514  0.001299   

       Voltage_0  Voltage_1  Voltage_2  Voltage_3  Identity  
0      

In [5]:
#shuffling
data_20=data_20.sample(frac=0.5)
data_30=data_30.sample(frac=0.5)
data_40=data_40.sample(frac=1)

In [6]:
data_20.head

<bound method NDFrame.head of        X_Value  Acceleration  Acceleration_0  Acceleration_1   Voltage  \
59634  0.98270     -0.485930        0.588825       -0.717198 -0.205831   
20680  0.03450     -2.003118       -0.689441        0.209122 -0.048939   
38857  0.94335      3.915657       -1.834982       -2.203720  0.202474   
19806  0.99030     -7.134656      -20.964168       13.143656 -0.117265   
41890  0.09550     -1.727880        0.376784       -0.138003  0.115214   
...        ...           ...             ...             ...       ...   
10825  0.54125     -4.006118      -18.228743        6.958145 -0.134763   
29066  0.45380      3.503476        2.812501       -1.831232  0.155894   
69727  0.48780     -1.160318       -2.767991       -2.858883  0.228189   
31811  0.59105      4.199991        4.247191        0.816131  0.158248   
90794  0.54160     -1.015139       -0.139384       -0.504732 -0.089951   

       Voltage_0  Voltage_1  Voltage_2  Voltage_3  Identity  
59634   0.101518   

In [7]:
data_30.head

<bound method NDFrame.head of        X_Value  Acceleration  Acceleration_0  Acceleration_1   Voltage  \
60852  0.04410     -2.041225       -5.486839       -1.864437 -0.128750   
36779  0.83945     -3.703959       -7.607748       -5.376242 -0.194124   
60303  0.01665     -2.116827       -3.981183       -2.513841 -0.228808   
14537  0.72685     -6.284357      -19.051371       -1.334621 -0.141765   
67196  0.36130      0.789209        4.090646       -0.206742  0.151947   
...        ...           ...             ...             ...       ...   
85122  0.25805     -1.839745        4.100836       -0.431828  0.108612   
6861   0.34305     -2.971550      -11.101341       -0.354511  0.116477   
54782  0.74010      7.225145        1.032063       -4.766906 -0.095526   
91274  0.56565      1.517931       -0.156206        3.390957 -0.190247   
98817  0.94280     10.632116       -8.850409        2.814090  0.038429   

       Voltage_0  Voltage_1  Voltage_2  Voltage_3  Identity  
60852  -0.117248   

In [8]:
import sklearn as sk
data_40.head

<bound method NDFrame.head of         X_Value  Acceleration  Acceleration_0  Acceleration_1   Voltage  \
4371    0.21855    -12.246894        5.607843        6.480159  0.039768   
15047   0.25235     -1.691247       -3.282686        3.946380 -0.193251   
20467   0.02335      0.829161        0.060134       -1.319428  0.219187   
33860   4.49300      1.376071       -8.250380       -1.268824  0.134400   
6114    0.30570      5.649320       -1.844436        1.765976 -0.049300   
...         ...           ...             ...             ...       ...   
9693    0.48465      3.732861       15.904452       -8.633555  0.197477   
29789  29.98945     -0.568415       -3.583620       -1.479941  0.153220   
20058   0.00290      4.720226        4.332277       12.912321  0.212272   
32483   4.42415     -1.356881        1.035255        9.532726 -0.162984   
27063   3.05315     -6.649701       -5.447304       -6.385514  0.171935   

       Voltage_0  Voltage_1  Voltage_2  Voltage_3  Comment  
4371   -

In [9]:
# X and y for training 20Hz dataset
dataset_20=data_20.values
X_20= dataset_20[:,0:9]
print(X_20)

y_20=dataset_20[:,9]
print(y_20)

[[ 9.827000e-01 -4.859300e-01  5.888250e-01 ...  1.043790e-01
   1.200000e-04  5.200700e-02]
 [ 3.450000e-02 -2.003118e+00 -6.894410e-01 ... -1.638790e-01
  -2.920000e-04  1.022790e-01]
 [ 9.433500e-01  3.915657e+00 -1.834982e+00 ... -1.798320e-01
  -5.500000e-04  8.662100e-02]
 ...
 [ 4.878000e-01 -1.160318e+00 -2.767991e+00 ... -1.242820e-01
  -1.204000e-03 -2.699100e-02]
 [ 5.910500e-01  4.199991e+00  4.247191e+00 ... -2.263480e-01
  -3.140000e-04 -3.305000e-03]
 [ 5.416000e-01 -1.015139e+00 -1.393840e-01 ... -1.359460e-01
  -6.420000e-04 -1.770300e-02]]
[0. 2. 2. ... 3. 2. 4.]


In [10]:
# X and y for training 30Hz dataset
dataset_30=data_30.values
X_30= dataset_30[:,0:9]
print(X_30)

y_30=dataset_30[:,9]
print(y_30)

[[ 4.4100000e-02 -2.0412250e+00 -5.4868390e+00 ...  2.4424600e-01
   2.5520000e-03 -5.1959000e-02]
 [ 8.3945000e-01 -3.7039590e+00 -7.6077480e+00 ...  2.1076000e-01
   1.2990000e-03 -2.2926300e-01]
 [ 1.6650000e-02 -2.1168270e+00 -3.9811830e+00 ...  1.1658000e-01
  -1.3440000e-03 -3.2077000e-02]
 ...
 [ 7.4010000e-01  7.2251450e+00  1.0320630e+00 ... -1.4998300e-01
   1.2800000e-04  2.7207700e-01]
 [ 5.6565000e-01  1.5179310e+00 -1.5620600e-01 ...  1.8974600e-01
  -7.4500000e-04 -1.1916200e-01]
 [ 9.4280000e-01  1.0632116e+01 -8.8504090e+00 ... -2.0764000e-01
  -6.2500000e-04 -4.0960000e-02]]
[3. 2. 3. ... 0. 4. 4.]


In [11]:
# X and y for training 30Hz dataset
dataset_40=data_40.values
X_40= dataset_40[:,0:9]
print(X_40)

y_40=dataset_40[:,9]
print(y_40)

[[ 2.1855000e-01 -1.2246894e+01  5.6078430e+00 ...  1.8120900e-01
  -3.4300000e-04 -3.7463100e-01]
 [ 2.5235000e-01 -1.6912470e+00 -3.2826860e+00 ...  2.1747700e-01
  -4.3400000e-04 -1.6880000e-01]
 [ 2.3350000e-02  8.2916100e-01  6.0134000e-02 ... -2.3077900e-01
   1.0600000e-04  6.1880000e-02]
 ...
 [ 2.9000000e-03  4.7202260e+00  4.3322770e+00 ... -3.0710000e-03
   1.4800000e-04  2.5047000e-02]
 [ 4.4241500e+00 -1.3568810e+00  1.0352550e+00 ...  2.1256800e-01
  -9.8400000e-04  2.9293900e-01]
 [ 3.0531500e+00 -6.6497010e+00 -5.4473040e+00 ...  6.9775000e-02
   1.1500000e-04 -5.3281100e-01]]
[1. 2. 0. ... 0. 3. 0.]


In [12]:
#Training random forest classifier

from sklearn.ensemble import RandomForestClassifier
rf_20 = RandomForestClassifier(n_estimators = 1000, random_state = 42)
rf_20.fit(X_20, y_20);

In [13]:
#Training random forest Classifier

from sklearn.ensemble import RandomForestClassifier
rf_30 = RandomForestClassifier(n_estimators = 1000, random_state = 42)
rf_30.fit(X_30, y_30);

In [14]:
#Training random forest Classifier

from sklearn.ensemble import RandomForestClassifier
rf_40 = RandomForestClassifier(n_estimators = 1000, random_state = 42)
rf_40.fit(X_40, y_40);

In [15]:
test_15= pd.read_csv(r'D:\Acads\BTP\Train\Train15Hz.csv')
test_15.head

<bound method NDFrame.head of        X_Value  Acceleration  Acceleration_0  Acceleration_1   Voltage  \
0      0.00000     -1.525662       -2.224441       -0.555582 -0.021681   
1      0.00005     -1.392776       -2.294180       -0.066691 -0.019098   
2      0.00010     -0.174552       -1.000812        0.584061 -0.019375   
3      0.00015     -0.383162       -1.101492        1.278678 -0.016724   
4      0.00020      1.119027        0.109123        2.087247 -0.017092   
...        ...           ...             ...             ...       ...   
99949  0.99925     -0.086289        1.216725       -1.050722 -0.210134   
99950  0.99930      0.034796        1.739032       -1.707600 -0.214406   
99951  0.99935     -0.306700        1.865864       -2.614070 -0.209294   
99952  0.99940     -0.349602        2.158573       -2.790634 -0.213611   
99953  0.99945     -0.223231        2.109338       -2.928357 -0.208421   

       Voltage_0  Voltage_1  Voltage_2  Voltage_3  Identity  
0       0.204297  -

In [16]:
#shuffling train data
test_15= test_15.sample(frac=1)
test_15.head

<bound method NDFrame.head of        X_Value  Acceleration  Acceleration_0  Acceleration_1   Voltage  \
19063  0.95315     -2.593422      -10.050955       -0.149398  0.097508   
78080  0.90540     -0.521580       -0.788279       -0.645028 -0.189361   
77550  0.87890     -0.429875        1.409121       -0.473732  0.224377   
16022  0.80110     -1.691739       -2.231808       -1.763106  0.138201   
18624  0.93120      5.178750       -1.091178       -1.183298  0.106479   
...        ...           ...             ...             ...       ...   
38661  0.93350      0.970160        0.420003       -1.289776  0.000308   
37607  0.88080     -2.141658       -4.871218       -1.877915  0.204580   
42954  0.14865     -0.519121        0.523015       -0.790838  0.224382   
3852   0.19260      0.549252      -12.141045        2.499190 -0.113152   
96677  0.83565     -3.601683       -3.667725       -2.412755  0.206820   

       Voltage_0  Voltage_1  Voltage_2  Voltage_3  Identity  
19063  -0.205614   

In [17]:
#Assigning X and Y parameter for testing 15Hz 

dataset_test_15 = test_15.values
X_test_15 = dataset_test_15[:,0:9]
print(X_test_15)

y_test_15= dataset_test_15[:,9]
print(y_test_15)

[[ 9.5315000e-01 -2.5934220e+00 -1.0050955e+01 ...  1.0773300e-01
  -3.9800000e-04  3.2082000e-02]
 [ 9.0540000e-01 -5.2158000e-01 -7.8827900e-01 ...  2.1597600e-01
  -1.0960000e-03  2.5870000e-03]
 [ 8.7890000e-01 -4.2987500e-01  1.4091210e+00 ... -1.3143500e-01
   2.8260000e-03 -6.0625000e-02]
 ...
 [ 1.4865000e-01 -5.1912100e-01  5.2301500e-01 ... -1.6420500e-01
   3.7100000e-04 -1.5601000e-01]
 [ 1.9260000e-01  5.4925200e-01 -1.2141045e+01 ... -9.0650000e-02
  -3.1800000e-04 -8.7306000e-02]
 [ 8.3565000e-01 -3.6016830e+00 -3.6677250e+00 ... -1.9138000e-02
   1.0120000e-03  4.0986000e-02]]
[1. 3. 3. ... 0. 1. 4.]


In [18]:
#20Hz testing 
test_20= pd.read_csv(r'D:\Acads\BTP\Train\Train20Hz.csv')
test_20.head
test_20= test_20.sample(frac=1)
test_20.head

<bound method NDFrame.head of        X_Value  Acceleration  Acceleration_0  Acceleration_1   Voltage  \
33213  0.66115      1.283998        0.748317       -2.018333 -0.044147   
8007   0.40035     -0.997929       13.738974       -2.075800  0.095613   
85486  0.27620      3.220248       -5.627668        2.038603 -0.151908   
88357  0.41975     -5.704502        3.557533        0.704752 -0.017648   
89584  0.48110      1.163405       -4.654634        1.852114 -0.226910   
...        ...           ...             ...             ...       ...   
63974  0.20015     -0.224338       -3.264023        0.123474  0.027998   
50658  0.53390     -1.986030       -3.271144       -2.166717 -0.015556   
87375  0.37065      2.067546        2.580938       -0.715605 -0.023306   
28883  0.44465     -3.562714       -1.784151       -0.643803  0.203045   
75415  0.77220     -1.297875       -1.737494       -3.056645 -0.073852   

       Voltage_0  Voltage_1  Voltage_2  Voltage_3  Identity  
33213  -0.172199   

In [19]:
#Assigning X and Y parameter for testing 20Hz

dataset_test_20 = test_20.values
X_test_20 = dataset_test_20[:,0:9]
print(X_test_20)

y_test_20= dataset_test_20[:,9]
print(y_test_20)

[[ 6.6115000e-01  1.2839980e+00  7.4831700e-01 ...  2.1676500e-01
  -2.0000000e-04  3.2348000e-02]
 [ 4.0035000e-01 -9.9792900e-01  1.3738974e+01 ... -2.0820300e-01
  -2.4600000e-04 -3.5642000e-02]
 [ 2.7620000e-01  3.2202480e+00 -5.6276680e+00 ...  2.1748700e-01
  -7.2900000e-04 -8.9860000e-03]
 ...
 [ 3.7065000e-01  2.0675460e+00  2.5809380e+00 ...  2.0485600e-01
   1.4230000e-03  5.3092000e-02]
 [ 4.4465000e-01 -3.5627140e+00 -1.7841510e+00 ... -1.7894100e-01
  -3.9300000e-04  5.8450000e-02]
 [ 7.7220000e-01 -1.2978750e+00 -1.7374940e+00 ... -1.5497300e-01
  -7.5600000e-04 -4.3698000e-02]]
[2. 1. 4. ... 4. 2. 3.]


In [20]:
# 30Hz testing
test_30= pd.read_csv(r'D:\Acads\BTP\Train\Train30Hz.csv')
test_30.head
test_30= test_30.sample(frac=1)
test_30.head

<bound method NDFrame.head of        X_Value  Acceleration  Acceleration_0  Acceleration_1   Voltage  \
64800  0.24150      3.826164        5.611403       -0.297535 -0.030985   
24535  0.22725      8.600717        2.520284       -4.956458  0.212981   
97131  0.85850     -0.906716        7.601797        1.900513 -0.008851   
69603  0.48165     -2.655131       -5.522568        1.529373 -0.238595   
38236  0.91230     -5.260976      -12.083707       -1.396131 -0.171348   
...        ...           ...             ...             ...       ...   
58062  0.90410     -4.057257       -0.394646        2.425672 -0.183647   
4957   0.24785    -17.684893      -23.149781        5.176695  0.224570   
96518  0.82785      0.993640       -2.030694       -0.894742 -0.109565   
38816  0.94130     -2.644191       -7.365134       -3.370442 -0.246657   
42843  0.14315      0.048440        0.166829       -0.560728  0.028943   

       Voltage_0  Voltage_1  Voltage_2  Voltage_3  Identity  
64800  -0.213086   

In [21]:
#Assigning X and Y parameter for testing 30Hz

dataset_test_30 = test_30.values
X_test_30 = dataset_test_30[:,0:9]
print(X_test_30)

y_test_30= dataset_test_30[:,9]
print(y_test_30)

[[ 2.415000e-01  3.826164e+00  5.611403e+00 ...  2.422500e-01
  -7.620000e-04  2.372350e-01]
 [ 2.272500e-01  8.600717e+00  2.520284e+00 ... -1.427310e-01
   1.116000e-03  1.349330e-01]
 [ 8.585000e-01 -9.067160e-01  7.601797e+00 ...  2.035360e-01
  -4.620000e-04  6.437700e-02]
 ...
 [ 8.278500e-01  9.936400e-01 -2.030694e+00 ...  2.184390e-01
  -4.520000e-04 -1.593490e-01]
 [ 9.413000e-01 -2.644191e+00 -7.365134e+00 ...  1.311530e-01
  -3.660000e-04 -5.778800e-02]
 [ 1.431500e-01  4.844000e-02  1.668290e-01 ... -2.262730e-01
   9.444542e-05  2.321230e-01]]
[3. 2. 4. ... 4. 2. 0.]


In [22]:
# 40Hz testing
test_40= pd.read_csv(r'D:\Acads\BTP\Train\Train40Hz.csv')
test_40.head
test_40= test_40.sample(frac=1)
test_40.head

<bound method NDFrame.head of         X_Value  Acceleration  Acceleration_0  Acceleration_1   Voltage  \
21128   0.05640      2.612366       -8.866247        2.753561  0.018387   
45893   5.09465      6.031629       -6.319045       -0.130651  0.032650   
40800   0.14000     -3.388524       -0.058718       -6.414063 -0.214142   
16855   0.34275    -10.031841       -4.525346        1.950505  0.217635   
9461    0.47305      8.634277      -12.592140      -13.022178 -0.225671   
...         ...           ...             ...             ...       ...   
16420   0.32100    -19.081240      -15.465815       -1.115418  0.141275   
38567  25.22835     -0.986373        3.363171       -3.187139 -0.150770   
17120   0.35600     -2.792319       -2.122779       -0.910181 -0.226147   
27851   3.09255      6.863611       -5.009591       -3.687670 -0.052716   
21952   0.09760     -3.856391       -2.906978       -3.956130  0.185013   

       Voltage_0  Voltage_1  Voltage_2  Voltage_3  Comment  
21128  -

In [23]:
#Assigning X and Y parameter for testing 40Hz

dataset_test_40 = test_40.values
X_test_40 = dataset_test_40[:,0:9]
print(X_test_40)

y_test_40= dataset_test_40[:,9]
print(y_test_40)

[[ 5.640000e-02  2.612366e+00 -8.866247e+00 ...  2.205720e-01
   9.320000e-05  2.152160e-01]
 [ 5.094650e+00  6.031629e+00 -6.319045e+00 ... -2.083760e-01
  -4.470000e-04  1.382000e-02]
 [ 1.400000e-01 -3.388524e+00 -5.871800e-02 ...  3.883200e-02
  -5.400000e-04  1.009360e-01]
 ...
 [ 3.560000e-01 -2.792319e+00 -2.122779e+00 ...  5.429600e-02
  -2.900000e-04  7.919400e-02]
 [ 3.092550e+00  6.863611e+00 -5.009591e+00 ... -1.967180e-01
   3.040000e-04 -5.550000e-04]
 [ 9.760000e-02 -3.856391e+00 -2.906978e+00 ... -2.457230e-01
   5.870000e-05 -7.685700e-02]]
[0. 4. 4. ... 2. 0. 0.]


In [23]:
# Testing mix data from different rotational speeds of 10Hz, 15Hz, 20Hz, 25Hz, 30Hz with 200 dataset row per fault
# for each rotational speed

test_mix= pd.read_csv(r'D:\Acads\BTP\Train\Mix_test_10_15_20_25_30.csv')
test_mix.head
test_mix= test_mix.sample(frac=1)
test_mix.head

<bound method NDFrame.head of       X_Value  Acceleration  Acceleration_0  Acceleration_1   Voltage  \
201   0.00010      0.767451        0.928313       -0.609862  0.193713   
1821  0.33655     -0.905486        0.068851       -0.702617 -0.043270   
2498  0.99090     -2.962822       -0.458368       -1.492317 -0.146686   
3198  0.04995     -2.365512       -4.791656       -0.557052 -0.023397   
2263  0.98960      0.678696        2.277424       -0.735945  0.218469   
...       ...           ...             ...             ...       ...   
3240  0.05205     -1.648468       -1.399480       -2.350388  0.056029   
446   0.05235     -0.849922       -0.125388       -0.526665  0.175490   
4387  0.50800    -11.522474       -6.219838       -4.512780 -0.208412   
2301  0.99150     -2.758760       -1.949782       -1.858678  0.183825   
3904  0.50485     -3.339476        0.601595        1.148552  0.134296   

      Voltage_0  Voltage_1  Voltage_2  Voltage_3  Identity  
201   -0.029580  -0.165588   0.0

In [24]:
#Assigning X and Y parameter for Mix testing  

dataset_test_mix = test_mix.values
X_test_mix = dataset_test_mix[:,0:9]
print(X_test_mix)

y_test_mix= dataset_test_mix[:,9]
print(y_test_mix)

[[ 1.0000000e-04  7.6745100e-01  9.2831300e-01 ... -1.6558800e-01
   1.3680000e-03 -4.7367000e-02]
 [ 3.3655000e-01 -9.0548600e-01  6.8851000e-02 ... -1.7581500e-01
  -4.0000000e-04  2.0943000e-02]
 [ 9.9090000e-01 -2.9628220e+00 -4.5836800e-01 ... -5.5848000e-02
   1.2300000e-04  3.7119000e-02]
 ...
 [ 5.0800000e-01 -1.1522474e+01 -6.2198380e+00 ...  1.8349700e-01
  -5.3000000e-04 -2.4454000e-02]
 [ 9.9150000e-01 -2.7587600e+00 -1.9497820e+00 ...  4.2605000e-02
   3.6720000e-03 -2.6713000e-02]
 [ 5.0485000e-01 -3.3394760e+00  6.0159500e-01 ...  1.0599600e-01
  -1.0660000e-03  1.7539100e-01]]
[2. 2. 0. ... 2. 3. 3.]


In [25]:
import numpy as np
#Predicting with 30Hz training
predictions_15 = rf_30.predict(X_test_15)
errors_15 = abs(predictions_15 - y_test_15)
print('Mean Absolute Error 15Hz with 30Hz:', round(np.mean(errors_15), 3), 'degrees.')

accuracy = 100 - np.mean(errors_15)
print('Accuracy:', round(accuracy, 3), '%.')

Mean Absolute Error 15Hz with 30Hz: 0.612 degrees.
Accuracy: 99.388 %.


In [26]:
predictions_20 = rf_30.predict(X_test_20)
errors_20 = abs(predictions_20 - y_test_20)
print('Mean Absolute Error 20Hz with 30Hz:', round(np.mean(errors_20), 3), 'degrees.')

accuracy = 100 - np.mean(errors_20)
print('Accuracy:', round(accuracy, 3), '%.')

Mean Absolute Error 20Hz with 30Hz: 0.276 degrees.
Accuracy: 99.724 %.


In [27]:
predictions_30 = rf_30.predict(X_test_30)
errors_30 = abs(predictions_30 - y_test_30)
print('Mean Absolute Error 30Hz with 30Hz:', round(np.mean(errors_30), 3), 'degrees.')

accuracy = 100 - np.mean(errors_30)
print('Accuracy:', round(accuracy, 3), '%.')

Mean Absolute Error 30Hz with 30Hz: 0.035 degrees.
Accuracy: 99.965 %.


In [28]:
predictions_40 = rf_30.predict(X_test_40)
errors_40 = abs(predictions_40 - y_test_40)
print('Mean Absolute Error 40Hz with 30Hz:', round(np.mean(errors_40), 3), 'degrees.')

accuracy = 100 - np.mean(errors_40)
print('Accuracy:', round(accuracy, 3), '%.')

Mean Absolute Error 40Hz with 30Hz: 0.192 degrees.
Accuracy: 99.808 %.


In [57]:
predictions_mix = rf_30.predict(X_test_mix)
errors_mix = abs(predictions_mix - y_test_mix)
print('Mean Absolute Error Mix Dataset with 30Hz:', round(np.mean(errors_mix), 3), 'degrees.')

accuracy = 100 - np.mean(errors_mix)
print('Accuracy:', round(accuracy, 3), '%.')

Mean Absolute Error Mix Dataset with 30Hz: 0.2 degrees.
Accuracy: 99.705 %.


In [29]:
#Predicting with 20Hz training
predictions_15 = rf_20.predict(X_test_15)
errors_15 = abs(predictions_15 - y_test_15)
print('Mean Absolute Error 15Hz with 20Hz:', round(np.mean(errors_15), 3), 'degrees.')

accuracy = 100 - np.mean(errors_15)
print('Accuracy:', round(accuracy, 3), '%.')

Mean Absolute Error 15Hz with 20Hz: 0.429 degrees.
Accuracy: 99.571 %.


In [30]:
predictions_20 = rf_20.predict(X_test_20)
errors_20 = abs(predictions_20 - y_test_20)
print('Mean Absolute Error 20Hz with 20Hz:', round(np.mean(errors_20), 3), 'degrees.')

accuracy = 100 - np.mean(errors_20)
print('Accuracy:', round(accuracy, 3), '%.')

Mean Absolute Error 20Hz with 20Hz: 0.027 degrees.
Accuracy: 99.973 %.


In [31]:
predictions_30 = rf_20.predict(X_test_30)
errors_30 = abs(predictions_30 - y_test_30)
print('Mean Absolute Error 20Hz with 30Hz:', round(np.mean(errors_30), 3), 'degrees.')

accuracy = 100 - np.mean(errors_30)
print('Accuracy:', round(accuracy, 3), '%.')

Mean Absolute Error 20Hz with 30Hz: 0.272 degrees.
Accuracy: 99.728 %.


In [32]:
predictions_40 = rf_20.predict(X_test_40)
errors_40 = abs(predictions_40 - y_test_40)
print('Mean Absolute Error 40Hz with 30Hz:', round(np.mean(errors_40), 3), 'degrees.')

accuracy = 100 - np.mean(errors_40)
print('Accuracy:', round(accuracy, 3), '%.')

Mean Absolute Error 40Hz with 30Hz: 0.496 degrees.
Accuracy: 99.504 %.


In [61]:
predictions_mix = rf_20.predict(X_test_mix)
errors_mix = abs(predictions_mix - y_test_mix)
print('Mean Absolute Error Mix Dataset with 30Hz:', round(np.mean(errors_mix), 3), 'degrees.')

accuracy = 100 - np.mean(errors_mix)
print('Accuracy:', round(accuracy, 3), '%.')

Mean Absolute Error Mix Dataset with 30Hz: 0.029 degrees.
Accuracy: 99.724 %.


## Results testing different datasets against 30Hz trained model
### 15Hz : 0.612 Mean Absolute error  
### 20Hz : 0.276 Mean Absolute error 
### 30Hz : 0.035 Mean Absolute error 
### 40Hz:  0.192 MAE
### Mix speed: 

## Results testing different datasets against 20Hz trained model
### 15Hz : 0.429 Mean Absolute error 
### 20Hz : 0.027 Mean Absolute error 
### 30Hz : 0.272 Mean Absolute error 
### 40Hz : 0.496 MAE
### Mix speed: 

In [33]:
#Predicting with 40Hz training
predictions_15 = rf_40.predict(X_test_15)
errors_15 = abs(predictions_15 - y_test_15)
print('Mean Absolute Error 15Hz with 40Hz:', round(np.mean(errors_15), 3), 'degrees.')

accuracy = 100 - np.mean(errors_15)
print('Accuracy:', round(accuracy, 3), '%.')

Mean Absolute Error 15Hz with 40Hz: 0.938 degrees.
Accuracy: 99.062 %.


In [34]:
predictions_20 = rf_40.predict(X_test_20)
errors_20 = abs(predictions_20 - y_test_20)
print('Mean Absolute Error 20Hz with 40Hz:', round(np.mean(errors_20), 3), 'degrees.')

accuracy = 100 - np.mean(errors_20)
print('Accuracy:', round(accuracy, 3), '%.')

Mean Absolute Error 20Hz with 40Hz: 0.802 degrees.
Accuracy: 99.198 %.


In [35]:
predictions_30 = rf_40.predict(X_test_30)
errors_30 = abs(predictions_30 - y_test_30)
print('Mean Absolute Error 20Hz with 40Hz:', round(np.mean(errors_30), 3), 'degrees.')

accuracy = 100 - np.mean(errors_30)
print('Accuracy:', round(accuracy, 3), '%.')

Mean Absolute Error 20Hz with 40Hz: 0.732 degrees.
Accuracy: 99.268 %.


In [36]:
predictions_40 = rf_40.predict(X_test_40)
errors_40 = abs(predictions_40 - y_test_40)
print('Mean Absolute Error 40Hz with 40Hz:', round(np.mean(errors_40), 3), 'degrees.')

accuracy = 100 - np.mean(errors_40)
print('Accuracy:', round(accuracy, 3), '%.')

Mean Absolute Error 40Hz with 40Hz: 0.0 degrees.
Accuracy: 100.0 %.


## Results testing different datasets against 40Hz trained model
### 15Hz : 0.938 Mean Absolute error 
### 20Hz : 0.802 Mean Absolute error 
### 30Hz : 0.732 Mean Absolute error 
### Mix speed: 